app/Http/Controllers/StatementController.php

```diff
@@ -176,5 +178,120 @@ public function getStatementHistory(Request $request, Validate $validate)
         }
     }
 
- 
+    public function editStatement($id)
+    {
+        $statement = [];
+        try {
+            $statement = Statement::where('id', $id)->first();
+            if ($statement) {
+                $filter['topicNum'] = $statement->topic_num;
+                $filter['campNum'] = $statement->camp_num;
+                $filter['asOf'] = 'default';
+                $topic = Camp::getAgreementTopic($filter);
+                $camp = Camp::getLiveCamp($filter);
+                $parentcampnum = isset($camp->parent_camp_num) ? $camp->parent_camp_num : 0;
+                $parentcamp = Camp::campNameWithAncestors($camp, $filter);
+                $nickName = Nickname::topicNicknameUsed($statement->topic_num);
+                $statement->go_live_time = date('m/d/Y, h:i:s A', $statement->go_live_time);
+                $WikiParser = new wikiParser;
+                $statement->parsed_value = $WikiParser->parse($statement->value);
+                $data = new stdClass();
+                $data->statement = $statement;
+                $data->topic = $topic;
+                $data->parent_camp = $parentcamp;
+                $data->nick_name = $nickName;
+                $data->parentcampnum = $parentcampnum;
+                $response[0]=$data;
+                $indexs = ['statement', 'topic', 'parent_camp', 'nick_name', 'parentcampnum'];
+                $response = $this->resourceProvider->jsonResponse($indexs, $response);
+                $response=$response[0];
+            }
+            return $this->resProvider->apiJsonResponse(200, trans('message.success.success'), $response, '');
+        } catch (Exception $e) {
+            return $this->resProvider->apiJsonResponse(400, trans('message.error.exception'), '', $e->getMessage());
+        }
+    }
+
+    public function storeStatement(Request $request, Validate $validate)
+    {
+        $validationErrors = $validate->validate($request, $this->rules->getStatementStoreValidationRules(), $this->validationMessages->getStatementStoreValidationMessages());
+        if ($validationErrors) {
+            return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+        }
+        $statement = [];
+        try {
+            $all = $request->all();
+            $filters['topicNum']=$all['topic_num'];
+            $filters['campNum']=$all['camp_num'];
+            $filters['asOf']='default';
+            $totalSupport =  Support::getAllSupporters($all['topic_num'], $all['camp_num'],0);
+            $loginUserNicknames =  Nickname::personNicknameIds();
+            $statement = new Statement();
```

Comment:
We defined a new statement object here. In that case we need to remove `$statement = []` (array) from line 221